### PR TITLE
protobuf v4.22.5

### PR DESCRIPTION
**Pull request by h-vetinari**
After https://github.com/conda-forge/libprotobuf-feedstock/pull/144 was finally merged...

Closes #180

### CODE REVIEW

**Comment by h-vetinari** [comment]
OK, it looks like we're running into https://github.com/pypa/pip/issues/12010

**Comment by h-vetinari** [comment]
Ugh...

```
  $PREFIX/include/absl/strings/cord.h:999:18: error: cannot convert 'absl::lts_20230125::Cord::HashFragmented<absl::lts_20230125::hash_internal::MixingHashState>(absl::lts_20230125::hash_internal::MixingHashState) const::<lambda(int)>' to 'int'
    999 |     ForEachChunk([&combiner, &hash_state](absl::string_view chunk) {
        |                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        |                  |
        |                  absl::lts_20230125::Cord::HashFragmented<absl::lts_20230125::hash_internal::MixingHashState>(absl::lts_20230125::hash_internal::MixingHashState) const::<lambda(int)>
   1000 |       hash_state = combiner.add_buffer(std::move(hash_state), chunk.data(),
        |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   1001 |                                        chunk.size());
        |                                        ~~~~~~~~~~~~~~
   1002 |     });
        |     ~
```

File: recipe/meta.yaml
Comment (xylar) [inline recipe/meta.yaml:6]: @h-vetinari, I think we want to keep this and just update it to 4.  There's no guarantee that the c++ and python version numbers will match in the future, right?

**Review by xylar — DISMISSED** [review]
Thanks for working on this @h-vetinari!  I think we want to keep separate `major` and `lib_major` variables in case they diverge again in the future.

File: recipe/meta.yaml
Comment (h-vetinari) [reply recipe/meta.yaml:6]: If they ever diverge again, we introduce the variable again. Not point keeping it around if that's not the case IMO.

Comment (xylar) [reply recipe/meta.yaml:6]: I disagree. I think the safer thing to do would be to keep `lib_major` as a reminder that the c++ and python version major versions may differ.  This has caused significant confusion and may do so in the future.

Comment (h-vetinari) [reply recipe/meta.yaml:6]: What's in this recipe or not will not reduce confusion about protobuf's unique versioning scheme and it's our job as maintainers to understand the pieces involved.

Also, this cycle of abseil/grpc/protobuf updates is exceptionally painful to pull off, so coming into a WIP PR with a big red "requested changes" based on a cosmetic (= not functionally relevant) preference is really... detrimental to me feeling like it's worth my free time to try to untangle this mess.

Comment (xylar) [reply recipe/meta.yaml:6]: I really do appreciate the work you put into this. It would help me as a maintainer if we keep the separate major version numbers but I will let it be if it's causing you anxiety.

**Review by xylar — COMMENTED** [review]
I don't think there's a way for me to rescind my request for changes with approving, which doesn't make sense to do while tests are failing. But you may ignore my request.

**Comment by xylar** [comment]
@h-vetinari, please remove me as a maintainer as part of this PR if you would. It seems like you've got this and my efforts are better spent elsewhere.

**Comment by h-vetinari** [comment]
> @h-vetinari, please remove me as a maintainer as part of this PR if you would. It seems like you've got this and my efforts are better spent elsewhere.

Did you notice that I followed your request for keeping the distinction between the two versions? I cannot stop you from removing yourself if you so choose, but saying that I disagree with you in something shouldn't have to be a reason to throw in the towel. Perhaps you'll reconsider?

**Comment by xylar** [comment]
> but saying that I disagree with you in something shouldn't have to be a reason to throw in the towel. Perhaps you'll reconsider?

It seemed from your initial response as if you would prefer that other maintainers not request changes to your PRs. I appreciate you listening to my request, but if you prefer to work alone, and that's certainly the impression that I got, I'm happy to oblige.

**Comment by xylar** [comment]
@h-vetinari, I see. Thank you for taking the time to explain.

In the team I work with, change request are a standard practice but I could see how they come across as aggressive. And I can see why you got annoyed, especially by my persistent.

I cared about this because I caused a not insignificant mess by misunderstanding the version in the past. My past mistakes are also part of why I'm not sure I can do more good than harm here.

I commented on a work in progress because it isn't standard practice to wait for a review on conda-forge -- it's typically necessary to react quickly or lose your chance.

As to the burden you're shouldering alone, I sincerely wish I could be of more help. I don't have the necessary expertise. I do hope you find folks willing to help who do.

**Review by xhochy — APPROVED** [review]

**Comment by h-vetinari** [comment]
> @h-vetinari, I see. Thank you for taking the time to explain.

Likewise! :)

> I don't have the necessary expertise. I do hope you find folks willing to help who do.

FWIW, most of what I know here is just the product of bashing my head against the wall for extended periods of time. 😅 
(not that I recommend much less insinuate you should do the same, but it can be done ;-))

**Pull request merged**
